testing/scenario_app/lib/main.dart

```diff
@@ -53,32 +22,39 @@ void main() {
     ..onMetricsChanged = _onMetricsChanged
     ..onPointerDataPacket = _onPointerDataPacket
     ..scheduleFrame();
+
   final ByteData data = ByteData(1);
   data.setUint8(0, 1);
   window.sendPlatformMessage('waiting_for_status', data, null);
 }
 
+void _handleDriverMessage(Map<String, dynamic> call) {
+  switch (call['method']) {
+    case 'set_scenario':
+      final Map<String, dynamic> scenarioData = call['args'];
+      loadScenario(scenarioData);
+      print('Loading scenario ${scenarioData['name']}');
+    break;
+    default:
+      print('Unimplemented method: ${call['method']}.');
+  }
+}
+
 Future<void> _handlePlatformMessage(
     String name, ByteData data, PlatformMessageResponseCallback callback) async {
-  print(name);
-  print(utf8.decode(data.buffer.asUint8List()));
-  if (name == 'set_scenario' && data != null) {
-    final String scenarioName = utf8.decode(data.buffer.asUint8List());
-    final Scenario candidateScenario = _scenarios[scenarioName];
-    if (candidateScenario != null) {
-      _currentScenario = candidateScenario;
-      window.scheduleFrame();
-    }
-    if (callback != null) {
-      final ByteData data = ByteData(1);
-      data.setUint8(0, candidateScenario == null ? 0 : 1);
-      callback(data);
-    }
-  } else if (name == 'write_timeline') {
-    final String timelineData = await _getTimelineData();
-    callback(Uint8List.fromList(utf8.encode(timelineData)).buffer.asByteData());
-  } else {
-    _currentScenario?.onPlatformMessage(name, data, callback);
+  print('$name = ${utf8.decode(data.buffer.asUint8List())}');
+
+  switch (name) {
+    case 'driver':
+      final Map<String, dynamic> call = json.decode(utf8.decode(data.buffer.asUint8List()));
+      _handleDriverMessage(call);
+    break;
+    case 'write_timeline':
+      final String timelineData = await _getTimelineData();
+      callback(Uint8List.fromList(utf8.encode(timelineData)).buffer.asByteData());
+    break;
+    default:
+      print('Unimplemented channel: $name.');
```

Comment:
throw for same reason

Reply:
Err, actually, why not just pass this down to `onPlatformMessage` of the scenario that's running?

Reply:
`getCurrentScenario()?.onPlatformMessage(....)`

Reply:
Done.

Reply:
This is an unimplemented channel. I think throwing should be fine.  If this case occurs, then some mistake was made in the platform test.

Reply:
But how will a scenario that wants to set up a custom channel do so now?

Reply:
If we want to do this, we should remove the `onPlatformMessage` from the scenario base class.

Reply:
oh gotcha. That makes sense. It should continue to call `currentScenario?.onPlatformMessage(..)`.